main/entry.c

```diff
@@ -1280,8 +1280,37 @@ static void makeTagEntriesForSubwords (tagEntryInfo *const subtag)
 	stringListDelete (list);
 }
 
-extern int makeTagEntry (const tagEntryInfo *const tag)
+static char *trimPrefixedWhitespaces (const char *name)
 {
+	const char *tmp = name;
+	size_t total_length = 0;
+	size_t prefix_length = 0;
+	bool enter_body = false;
+
+	for (tmp = name;
+		 *tmp != '\0';
+		 tmp++)
+	{
+		unsigned char c = *tmp;
+		if (c < '!' && !enter_body)
+			prefix_length++;
+		else
+			enter_body = true;
+
+		if (prefix_length == 0)
+			return NULL;
+
+		total_length++;
+	}
+	return eStrndup (name + prefix_length, total_length - prefix_length);
+}
```

Comment:
I'm not sure if I misunderstood this function, but it could possibly be implemented more simply like this:

```c
static char *trimPrefixedWhitespaces(const char *name)
{
  const char *start = name;
  for (; isspace(*start); start++)
    ;
  return eStrndup(start, strlen(start));
}
```

It would be surprising to me that the ctags code doesn't already have this kind of function though.

Reply:
I should do as you wrote.

At the first time, I used memmove not to allocate an extra memory.
So my trimPrefixedWhitespaces is a bit complicated.

Thank you for the suggestion.